File: go/problems/day19_test.go
package problems

import "testing"

var day19 Solvable[int] = Day19{}

func TestDay19_TestSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day19_1")
	result := day19.Solve(input)

	var expected int = 19114
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

func TestDay19_FullSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day19_2")
	result := day19.Solve(input)

	var expected int = 353046
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

File: ts/problems/day19.test.ts
import test, { describe } from "node:test";
import assert from "node:assert";

import solve from "./day19.js";
import { readProblemInput } from "../inputReader.js";

describe("Day 19", () => {
  test("test set should yield expected result", () => {
    const input = readProblemInput("day19_1");
    const result = solve(input);
    assert.equal(result, 19114);
  });

  test("full set should yield expected result", () => {
    const input = readProblemInput("day19_2");
    const result = solve(input);
    assert.equal(result, 353046);
  });
});
